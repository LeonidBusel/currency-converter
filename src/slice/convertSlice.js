import { createSlice } from '@reduxjs/toolkit'

export const convertSlice = createSlice({
    name: 'convert',
    initialState: {
        isFetching: false,
        convert: {},
        amount: 0
    },
    reducers: {
        convertFetch: state => {
            state.isFetching = true;
        },
        convertFetchSuccess: (state, { payload }) => {
            state.isFetching = false;
            state.convert = Object.assign({}, payload.convert);
            state.amount = payload.amount
        },
        convertFetchFail: (state) => {
            state.isFetching = false;
            state.convert = {};
            state.amount = 0;
        },
        convertClear: (state) => {
            state.isFetching = false;
            state.convert = {};
            state.amount = 0;
        }
    }
})

export const { convertFetch, convertFetchSuccess, convertFetchFail, convertClear } = convertSlice.actions

export default convertSlice.reducer