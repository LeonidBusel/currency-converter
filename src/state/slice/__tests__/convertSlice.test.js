import convertSlice, { initialState, convertFetch, convertFetchSuccess, convertFetchFail, convertClear } from "../convertSlice";

const modifyState = {
    "amount": 55,
    "convert": {
        "a": 1,
        "b": 2,
    },
    "isFetching": false
}

describe('convertSlice', () => {
    it('handle convertFetch correctly', () => {
        const actual = convertSlice(initialState, convertFetch());

        expect(actual).toMatchSnapshot();
    });

    it('handle convertFetchSuccess correctly', () => {
        const actual = convertSlice(initialState, convertFetchSuccess({ convert: { a: 1, b: 2 }, amount: 55 }));

        expect(actual).toMatchSnapshot();
    });

    it('handle convertFetchFail correctly', () => {
        const actual = convertSlice(modifyState, convertFetchFail());

        expect(actual).toMatchSnapshot();
    });

    it('handle convertClear correctly', () => {
        const actual = convertSlice(modifyState, convertClear());

        expect(actual).toMatchSnapshot();
    });
});