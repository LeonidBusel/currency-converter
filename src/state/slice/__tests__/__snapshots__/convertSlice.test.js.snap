// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertSlice handle convertClear correctly 1`] = `
Object {
  "amount": 0,
  "convert": Object {},
  "isFetching": false,
}
`;

exports[`convertSlice handle convertFetch correctly 1`] = `
Object {
  "amount": 0,
  "convert": Object {},
  "isFetching": true,
}
`;

exports[`convertSlice handle convertFetchFail correctly 1`] = `
Object {
  "amount": 0,
  "convert": Object {},
  "isFetching": false,
}
`;

exports[`convertSlice handle convertFetchSuccess correctly 1`] = `
Object {
  "amount": 55,
  "convert": Object {
    "a": 1,
    "b": 2,
  },
  "isFetching": false,
}
`;
